source/components/recipe-search/recipe-search.html

```diff
@@ -2,42 +2,117 @@
+
+    <div id="filter-form">
+      <img id="close-button" src="static/common/close-icon.png" />
+      <div id="cost-form">
+        <h3>Cost</h3>
+        <div>
+          <label for="input-cost-min">Min</label>
+          <input type="number" id="input-cost-min" class="input-cost" />
+        </div>
+        <div>
+          <label for="input-cost-max">Max</label>
+          <input type="number" id="input-cost-max" class="input-cost" />
+        </div>
+      </div>
+
+      <div id="category-form">
+        <h3>Categories</h3>
+        <div>
+          <input
+            type="checkbox"
+            id="input-gluten-free"
+            class="input-category"
+          />
+          <label for="input-gluten-free">gluten-free</label>
+        </div>
+        <div>
+          <input type="checkbox" id="input-healthy" class="input-category" />
+          <label for="input-healthy">healthy</label>
+        </div>
+        <div>
+          <input
+            type="checkbox"
+            id="input-high-protein"
+            class="input-category"
+          />
+          <label for="input-high-protein">high-protein</label>
+        </div>
+        <div>
+          <input type="checkbox" id="input-vegan" class="input-category" />
+          <label for="input-vegan">vegan</label>
+        </div>
+        <div>
+          <input type="checkbox" id="input-vegetarian" class="input-category" />
+          <label for="input-vegetarian">vegetarian</label>
+        </div>
+      </div>
+
+      <!-- There are 5 kinds of sort, when default or click clear button, it is default sort. -->
+      <div id="sort-form">
+        <h3>Sort</h3>
+        <div>
+          <input
+            type="radio"
+            id="sort-time-descending"
+            class="input-sort"
+            name="sort-type"
+            value="sort-time-descending"
+          />
+          <label for="sort-time-descending">time ↓</label>
+        </div>
+        <div>
+          <input
+            type="radio"
+            id="sort-time-ascending"
+            class="input-sort"
+            name="sort-type"
+            value="sort-time-ascending"
+          />
+          <label for="sort-time-ascending">time ↑</label>
+        </div>
+        <div>
+          <input
+            type="radio"
+            id="sort-cost-descending"
+            class="input-sort"
+            name="sort-type"
+            value="sort-cost-descending"
+          />
+          <label for="sort-cost-descending">cost ↓</label>
+        </div>
+        <div>
+          <input
+            type="radio"
+            id="sort-cost-ascending"
+            class="input-sort"
+            name="sort-type"
+            value="sort-cost-ascending"
+          />
+          <label for="sort-cost-ascending">cost ↑</label>
+        </div>
+      </div>
+      <input type="submit" id="submit-button" />
+      <input type="reset" id="reset-button" />
+    </div>
+  </form>
+
+  <div id="recipe-cart-grid">
```

Comment:
Please rename to `recipe-card-grid`.
```suggestion
  <div id="recipe-card-grid">
```